src/payment/strategies/moneris/moneris-payment-strategy.ts

```diff
@@ -89,29 +141,74 @@ export default class MonerisPaymentStrategy implements PaymentStrategy {
         if (nonce !== undefined) {
             return this._store.dispatch(this._paymentActionCreator.submitPayment({
                 methodId: payment.methodId,
-                paymentData: { nonce },
+                paymentData: { nonce, shouldSaveInstrument, shouldSetAsDefaultInstrument },
             }));
         }
 
         return this._store.getState();
     }
 
-    finalize(): Promise<InternalCheckoutSelectors> {
-        return Promise.reject(new OrderFinalizationNotRequiredError());
-    }
+    protected async executeWithVaulted(payment: OrderPaymentRequestBody): Promise<InternalCheckoutSelectors> {
+        if (this._isHostedPaymentFormEnabled(payment.methodId)) {
+            const form = this._hostedForm;
 
-    deinitialize(): Promise<InternalCheckoutSelectors> {
-        if (this._windowEventListener) {
-            window.removeEventListener('message', this._windowEventListener);
-            this._windowEventListener = undefined;
+            if (!form) {
+                throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+            }
+
+            await form.validate();
+            await form.submit(payment);
+
+            return await this._store.dispatch(this._orderActionCreator.loadCurrentOrder());
```

Comment:
Why do we need to load the current order here?

Reply:
It's used to update the state of the checkout and check if the payment for the order has been submitted, neither `form.validate` and `form.submit` update the state.
This is what the base credit-card strategy does.